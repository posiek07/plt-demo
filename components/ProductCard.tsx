import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';
import { ProductDataType } from '../types';

const ProductCard: React.FC<ProductDataType> = ({ id, colour, price, img, name }) => {
  const navigation = useNavigation();
  return (
    // @ts-ignore
    <TouchableOpacity
      style={styles.container}
      onPress={() =>
        navigation.navigate('ProductScreen', {
          id: id,
        })
      }>
      <View style={styles.cardImageWrapper}>
        <Image source={{ uri: img }} style={styles.cardImage} />
      </View>
      <Text>{name}</Text>
      <Text>{colour}</Text>
      <Text>
        {price} {'\u00A3'}
      </Text>
    </TouchableOpacity>
  );
};

export default ProductCard;

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    width: '100%',
    backgroundColor: '#cccccc',
    justifyContent: 'center',
  },
  cardImageWrapper: {
    height: 200,
  },
  cardImage: {
    flex: 1,
    backgroundColor: '#cccccc',
    width: '100%',
    resizeMode: 'contain',
  },
  cardTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    paddingLeft: 5,
  },
});
