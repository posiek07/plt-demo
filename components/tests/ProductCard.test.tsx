import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../ProductCard';
jest.mock('@react-navigation/native');

const product = {
  id: 1,
  colour: 'Black',
  name: 'Black Sheet Strappy Textured Glitter Bodycon Dress',
  price: 10,
  img: 'http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024',
};

test('renders correctly', () => {
  const tree = renderer
    .create(
      <ProductCard
        id={product.id}
        colour={product.colour}
        name={product.name}
        price={product.price}
        img={product.img}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
