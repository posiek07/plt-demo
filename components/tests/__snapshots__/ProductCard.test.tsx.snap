// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#cccccc",
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": 200,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
        }
      }
      style={
        Object {
          "backgroundColor": "#cccccc",
          "flex": 1,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
  <Text>
    Black Sheet Strappy Textured Glitter Bodycon Dress
  </Text>
  <Text>
    Black
  </Text>
  <Text>
    10
     
    Â£
  </Text>
</View>
`;
